lib/vagrant-service-manager/services/docker.rb

```diff
@@ -71,6 +69,35 @@ def self.print_env_info(ui, options)
           PluginUtil.print_shell_configure_info(ui, ' docker')
         end
       end
+
+      def self.install_cli(machine, ui, options)
+        Installer.new('docker', machine, ui, options).run
+      end
+
+      def self.binary_path(version)
+        "#{BIN_FOLDER}/docker/#{version}/docker"
+      end
+
+      def self.binary_download_url(machine, version)
+        arch = machine.env.host.capability(:os_arch)
+        "#{BINARY_BASE_URL}/#{os_type}/#{arch}/docker-#{version}"
```

Comment:
> And how do you know that the CDK will have a download URL. The mechanism might be completely different in which case things get even more complicated. 

Probably we need to think it when there is requirement of implementing for CDK. Implementing with wild assumption is what I want to avoid here. At least, the code here assumes that there will be some download URL for CDK and hence I make it separate implementation class for CDK.